homeassistant/components/netatmo/camera.py

```diff
@@ -243,81 +198,97 @@ def is_on(self):
         """Update entity status."""
 
-        _LOGGER.debug("Updating camera netatmo '%s'", self._name)
-
-        # Refresh camera data.
+        # Refresh camera data
         self._data.update()
 
-        # URLs.
-        self._vpnurl, self._localurl = self._data.camera_data.cameraUrls(
-            camera=self._camera_name
-        )
+        camera = self._data.camera_data.get_camera(cid=self._camera_id)
 
-        # Identifier
-        self._id = self._data.camera_data.cameraByName(
-            camera=self._camera_name, home=self._home
-        )["id"]
-
-        # Monitoring status.
-        self._status = self._data.camera_data.cameraByName(
-            camera=self._camera_name, home=self._home
-        )["status"]
+        # URLs
+        self._vpnurl, self._localurl = self._data.camera_data.camera_urls(
+            cid=self._camera_id
+        )
 
-        _LOGGER.debug("Status of '%s' = %s", self._name, self._status)
+        # Monitoring status
+        self._status = camera.get("status")
 
         # SD Card status
-        self._sd_status = self._data.camera_data.cameraByName(
-            camera=self._camera_name, home=self._home
-        )["sd_status"]
+        self._sd_status = camera.get("sd_status")
 
         # Power status
-        self._alim_status = self._data.camera_data.cameraByName(
-            camera=self._camera_name, home=self._home
-        )["alim_status"]
+        self._alim_status = camera.get("alim_status")
 
         # Is local
-        self._is_local = self._data.camera_data.cameraByName(
-            camera=self._camera_name, home=self._home
-        )["is_local"]
-
-        # VPN URL
-        self._vpn_url = self._data.camera_data.cameraByName(
-            camera=self._camera_name, home=self._home
-        )["vpn_url"]
+        self._is_local = camera.get("is_local")
 
         self.is_streaming = self._alim_status == "on"
 
-        if self.model == "Presence":
-            # Light mode status
-            self._light_mode_status = self._data.camera_data.cameraByName(
-                camera=self._camera_name, home=self._home
-            )["light_mode_status"]
+    def turn_on(self):
+        """Instruct the light to turn on."""
+        _LOGGER.debug("Set the flood light on for the camera '%s'", self._name)
+        if self._set_mode("on"):
+            self.is_streaming = True
+            self.schedule_update_ha_state()
+
+    def turn_off(self):
+        """Instruct the light to turn off."""
+        _LOGGER.debug("Set the flood light off for the camera '%s'", self._name)
+        if self._set_mode("off"):
+            self.is_streaming = False
+            self.schedule_update_ha_state()
+
+    def _set_mode(self, mode: str):
+        """Set camera mode ('on', 'off')."""
+        try:
+            config = f'{{"mode":"{mode}"}}'
+            if self._localurl:
+                resp = requests.get(
+                    f"{self._localurl}/command/changestatus?status=" f"{config}",
```

Comment:
```suggestion
                    f"{self._localurl}/command/changestatus?status={config}",
```

Reply:
Also, why not use if…elif to set the url and have one piece of code to do the request ?